kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/MaterializedRowTrackingColumn.java

```diff
@@ -264,16 +281,23 @@ private static ColumnarBatch transformPhysicalRowId(
                 new ScalarExpression(
                     "ADD",
                     Arrays.asList(
-                        new Column(StructField.METADATA_ROW_INDEX_COLUMN_NAME),
+                        new Column(
+                            dataBatch
+                                .getSchema()
+                                .at(dataBatch.getSchema().indexOf(MetadataColumnType.ROW_INDEX))
+                                .getName()),
```

Comment:
We have to take this indirection because we don't know which name the engine decided to give to the row index metadata column.